src/fondant/pipeline/pipeline.py

```diff
@@ -362,16 +376,41 @@ def read(
                 msg,
             )
 
-        operation = ComponentOp(
-            name_or_path,
-            produces=produces,
-            arguments=arguments,
-            input_partition_rows=input_partition_rows,
-            resources=resources,
-            cache=cache,
-            cluster_type=cluster_type,
-            client_kwargs=client_kwargs,
-        )
+        if isinstance(ref, (str, Path)):
```

Comment:
Ideally I can flip this and check if the ref is a component.

Reply:
You can with the current implementation, right? Since it's a subclass of a component subclass?

Reply:
Resolved by instantiating (this might be more difficult later because of args)